nixos/modules/services/hardware/g810-led.nix

```diff
@@ -0,0 +1,45 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.g810-led;
+in
+{
+  options = {
+    services.g810-led = {
+      enable = lib.mkEnableOption "g810-led, a Linux LED controller for some Logitech G Keyboards";
+
+      package = lib.mkPackageOption pkgs "g810-led" { };
+
+      profile = lib.mkOption {
+        type = lib.types.nullOr lib.types.lines;
+        default = null;
+        example = ''
+          # G810-LED Profile (turn all keys on)
+
+          # Set all keys on
+          a ffffff
+
+          # Commit changes
+          c
+        '';
+        description = ''
+          Keyboard profile to apply at boot time.
+
+          The upstream repository provides [example configurations](https://github.com/MatMoul/g810-led/tree/master/sample_profiles).
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.etc."g810-led/profile" = lib.mkIf (cfg.profile != null) cfg.profile;
+
+    services.udev.packages = [ config.package ];
```

Comment:
I suppose you don't need to add this to `environment.systemPackages`? There are a few binaries there but I am not sure if they are useful (I won't be able to test this)

![image](https://github.com/user-attachments/assets/b152b6a7-a0f4-4624-b447-22afba447091)

Reply:
No, I don't think that it is in the scope of this nixos module.